includes/core/server/class-request.php

```diff
@@ -460,7 +460,7 @@ final public static function extend( Request $base_request ) {
 				'wcpay_core_extend_class_incorrectly'
 			);
 		}
-		$obj = new $current_class( $base_request->api_client, $base_request->http_interface );
+		$obj = new $current_class( $base_request->api_client, $base_request->http_interface, $base_request->id );
```

Comment:
Nitpick: I would do something like `$base_request->id ?? null` to avoid possible undefined id property since we are checking only if id is not null inside the function that sets id that is passed via constructor.

Reply:
Done in https://github.com/Automattic/woocommerce-payments/pull/7775/commits/4da980d40748d6656ecea999d67e69ea3accbd2a.